src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,203 @@
+package frc.trigon.robot.subsystems.arm;
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.*;
+import frc.trigon.robot.utilities.Conversions;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Supplier;
+
+public class Arm extends SubsystemBase {
+    private final static Arm INSTANCE = new Arm();
+
+    private final CANSparkMax
+            masterAngleMotor = ArmConstants.MASTER_ANGLE_MOTOR,
+            followerAngleMotor = ArmConstants.FOLLOWER_ANGLE_MOTOR,
+            masterElevatorMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+            followerElevatorMotor = ArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final CANcoder angleEncoder = ArmConstants.ANGLE_ENCODER;
+    private final TalonSRX elevatorEncoder = ArmConstants.ELEVATOR_ENCODER;
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGenerationTime,
+            lastElevatorMotorProfileGenerationTime;
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+    }
+
+    public Command getSetArmState(ArmConstants.ArmState targetState) {
+        if (getElevatorPositionRevolutions() < targetState.elevatorPosition) {
+            return new SequentialCommandGroup(
+                    setAngleStateDeferedCommand(targetState.angle),
+                    setElevatorPositionDeferedCommand(targetState.elevatorPosition)
+            );
+        }
+        return new SequentialCommandGroup(
+                setElevatorPositionDeferedCommand(targetState.elevatorPosition),
+                setAngleStateDeferedCommand(targetState.angle)
+        );
+    }
+
+    private Command setAngleStateDeferedCommand(Rotation2d targetAngle){
+        Supplier<Command> getSetTargetAngleCommandSupplier = () -> getSetTargetAngleCommand(targetAngle);
+        return new DeferredCommand(
+                getSetTargetAngleCommandSupplier,
+                Set.of(this)
+        );
+    }
+
+    private Command setElevatorPositionDeferedCommand(double targetPosition){
+        Supplier<Command> getSetTargetElevatorPositionCommandSupplier = () -> getSetTargetElevatorPositionCommand(targetPosition);
+        return new DeferredCommand(
+                getSetTargetElevatorPositionCommandSupplier,
+                Set.of(this)
+        );
```

Comment:
The reason you want to use a deferred command is the `getElevatorPositionRevolutions() < targetState.elevatorPosition` that should be inside a supplier, not the commands below. Rework this.

Reply:
okay, but how do I put `getElevatorPositionRevolutions() < targetState.elevatorPosition`  inside a supplier and in a DefferedCommand? Do I need to make a function that returns a command that checks and then make a Command that uses said function in a DefferedCommand? or is it something else?

Reply:
Yes, and to do what you said